collect_app/src/main/java/org/odk/collect/android/javarosawrapper/FormController.java

```diff
@@ -516,6 +516,9 @@ public boolean saveAnswer(FormIndex index, IAnswerData data) throws JavaRosaExce
         try {
             return formEntryController.saveAnswer(index, data, true);
         } catch (Exception e) {
+            Timber.w("Error saving answer of type %s with ref %s for index %s",
+                    data.getClass().toString(), index.getReference(), index);
```

Comment:
Ok sorry now I'm being really paranoid, but I think it'd be good to make that we account for `null` `data or `index` so there is no risk of ending up with a `NullPointerException` here.